### PR TITLE
Add support for type union and type intersection for nullable types

### PR DESCRIPTION
**Pull request by Cypher1**
This PR adds support for computing unions and intersections of nullable Schema types.

This is needed for resolving particles that make use nullable types, particularly of handles containing nullable inline schemas. 

#nullability

### CODE REVIEW

**Comment by Cypher1** [comment]
CC' @bgogul

File: src/types/tests/schema-test.ts [hunk 1]
```diff
@@ -682,14 +689,39 @@ describe('schema', () => {
...
+    assert.deepEqual(intersectionBA.toLiteral(), intersectT);
     assert.deepEqual(aType.toLiteral(), aTypeLit, 'input (a) to intersection should not be modified');
     assert.deepEqual(bType.toLiteral(), bTypeLit, 'input (b) to intersection should not be modified');
   };
+  const verifyIntersectOf = async (typeBuilder: (type: string) => string) => {
+    await verifyIntersectOfTypes(`${typeBuilder(`Foo {a: Text, b: Text}`)}`, `${typeBuilder(`Foo {a: Text}`)}`);
+  };
+  it('tests schema union, with nullable and non-nullable ints', async () => {
+    await verifyUnionOfTypes(`Int`, `Int?`);
+  });
```
Comment (mykmartin) [inline src/types/tests/schema-test.ts:709]: Not especially important, but you should be able to just pass the verify method directly:

`it('tests the thing', verifyUnionOfTypes('Int', 'Int?'));`

(That may or may not be more readable...)

Also, not sure why you're using interpolation literals; there doesn't seem to be any need for them.

**Review by mykmartin — APPROVED** [review]

File: src/types/tests/schema-test.ts (hunk 1 above)
Comment (Cypher1) [reply src/types/tests/schema-test.ts:709]: Thanks, that's neat. I'll switch the quotes too.

**Comment by Cypher1** [comment]
CC' @shans (removed from reviewer)

**Pull request closed without merging**